PrestaShop/PrestaShop
# Migrate hooks for new Order view page

| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Migrates olds hooks from legacy page to new one. See details below
| Type?         | improvement
| Category?     | CO
| BC breaks?    | yes, 2 hooks have been moved and renamed but aliases keep compatibility but some are really not compatible any more because of different parameters
| Deprecations? | no
| Fixed ticket? | Fixes https://github.com/PrestaShop/PrestaShop/issues/16523
| How to test?  | See below

### New hooks mapping

See below https://github.com/PrestaShop/PrestaShop/pull/16144#issuecomment-550356204

### How to test

1. Build assets. Reinstall shop (SQL file modified).
2. Go to modules/folder in CLI
3. Copy [test module](https://github.com/matks/orderhooks) there: `git clone git@github.com:matks/orderhooks.git`
4. Go to BO "Module catalog" and search for "orderhook"
5. Install "orderhook" test module
6. Go to 1 order view page, example `/admin-dev/index.php/sell/orders/orders/1/view`
7. See how hooks are displayed and compare to mapping
8. Refresh. I added a "random 50%" to display the 2 usecases explained below

### Known limitation

Mapping requires that "when there is not enough space for all buttons to be rendered, a dropdown list appears instead and buttons are put inside". I cant do that as my integration skills are very low.

What I did instead:
- if 1 or 2 buttons are added (like module test) they are displayed
- if more than 2 buttons are added, I display 1st button THEN next buttons are in dropdown (see screenshot below)

=> these 2 usecases can be seen with the test module, when you refresh there is a 50% luck you have usecase 1 or 2

### Screenshots

![Capture d’écran 2019-11-22 à 17 11 13](https://user-images.githubusercontent.com/3830050/69441572-4b953780-0d4b-11ea-9853-c7f7abf01841.png)
![Capture d’écran 2019-11-22 à 17 11 21](https://user-images.githubusercontent.com/3830050/69441585-4f28be80-0d4b-11ea-8548-2add34881e59.png)



---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/prestashop/prestashop/16144)
